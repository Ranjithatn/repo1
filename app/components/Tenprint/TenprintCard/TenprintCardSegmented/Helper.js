

function repositionBiometrics(biometrics) {
  const output = [];
  biometrics.forEach( item => {
    if ( item.name.indexOf('THUMB') != -1 ) { output[0] = item; }
    else if ( item.name.indexOf('INDEX') != -1 ) { output[1] = item; }
    else if ( item.name.indexOf('MIDDLE') != -1 ) { output[2] = item; }
    else if ( item.name.indexOf('RING') != -1 ) { output[3] = item; }
    else if ( item.name.indexOf('LITTLE') != -1 ) { output[4] = item; }
  });

  return output;
}


// function sortAndRepositionBiometrics(biometrics) {
//   const left = biometrics.filter(item => { return item.name.indexOf('L.') != -1; });
//   const right = biometrics.filter(item => { return item.name.indexOf('R.') != -1; });

//   const leftRepos = repositionBiometrics(left);
//   console.log("leftRepos",leftRepos);

//   // output = [...left,...right];
//   return [...left,...right];
// }


export default {
  repositionBiometrics
}
