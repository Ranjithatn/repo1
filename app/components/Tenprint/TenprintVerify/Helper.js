// this will filter all the biometrics which belongs to a specific hand and has a specific type
export const findAllFingers = (biometrics, hand, type) => {
  let output = [];

  biometrics &&
    biometrics.map(finger => {
      if (
        finger.position.indexOf(hand) != -1 &&
        (finger.impression === type || finger.type === type)
      ) {
        if (
          finger.position != "LeftFourFingers" &&
          finger.position != "RightFourFingers" &&
          finger.position != "BothThumbs"
        ) {
          output.push(finger);
        }
      }
    });

  return output;
};

// this will filter the three fingers of a given hand
export const findThreeFingers = biometrics => {
  let output = [];

  biometrics &&
    biometrics.map(finger => {
      if (
        finger.position === "LeftFourFingers" ||
        finger.position === "RightFourFingers" ||
        finger.position === "BothThumbs"
      ) {
        output.push(finger);
      }
    });

  return output;
};

export const findAllMatches = (hand, matches, user = 0) => {
  // console.log("findAllMatches::hand, matches, user",hand, matches, user);
  if (hand.length < 1) {
    return hand;
  }
  let matchFound = [];
  matchFound.length = 0;
  if (matches.length < 1) {
    return matchFound;
  }

  hand.forEach(finger => {
    let found = findMatchByPositionAndImpression(
      matches,
      finger.position,
      finger.impression,
      finger.id,
      user
    );

    if (found) {
      matchFound.push(found);
    }
    // else {
    //   if ( finger.annotation !== "NotAnnotated" ) {
    //     output.push({ ...finger, na: true, fingerprintPosition: finger.position });
    //   }
    // }
  });
  // console.log("myoutput",matchFound)
  return matchFound;
};

export const findMatchByPositionAndImpression = (
  matches,
  fingerprintPosition,
  impression,
  fingerId,
  user
) => {
  if (matches.length < 1) {
    return undefined;
  }
  let found;

  matches.forEach(data => {
    let currUser = data.matchInfos[user];

    if (
      currUser.fingerprintPosition === fingerprintPosition &&
      (currUser.impression === impression ||
        currUser.impression === "Unknown") &&
      data.probeBiometricsID === fingerId
    ) {
      found = currUser;
    }
  });

  return found;
};

export const sortFingers = (hand, type) => {
  let output = [];

  if (hand && hand.fingers) {
    hand = hand.fingers;
  }

  hand.map((finger, index) => {
    if (type !== "Palm") {
      if (finger.position.indexOf("Thumb") != -1) {
        output[0] = finger;
      } else if (finger.position.indexOf("Index") != -1) {
        output[1] = finger;
      } else if (finger.position.indexOf("Middle") != -1) {
        output[2] = finger;
      } else if (finger.position.indexOf("Ring") != -1) {
        output[3] = finger;
      } else if (finger.position.indexOf("Little") != -1) {
        output[4] = finger;
      } else if (finger.position.indexOf("Unknown Finger") != -1) {
        output.push(finger);
      } else if (finger.position.indexOf("Right Four Fingers") != -1) {
        output.push(finger);
      } else if (finger.position.indexOf("Left Four Fingers") != -1) {
        output.push(finger);
      }
    } else if (type === "Palm") {
      if (
        finger.position.indexOf("Lower") != -1 &&
        finger.position.indexOf("Palm") != -1
      ) {
        output[0] = finger;
      } else if (
        finger.position.indexOf("Writers") != -1 &&
        finger.position.indexOf("Palm") != -1
      ) {
        output[1] = finger;
      }
    }
  });

  hand = output;
  return hand;
};

export const sortFingerMatches = (hand, selected) => {
  if (hand && hand.fingers) {
    hand = hand.fingers;
  }

  let output = [];

  if (hand.length > 0) {
    hand.map((finger, index) => {
      const currFinger = finger;
      // console.log("currFinger, selected",currFinger, selected);

      if (selected.type !== "Palm") {
        if (currFinger.fingerprintPosition.indexOf("Thumb") != -1) {
          output[0] = finger;
        } else if (currFinger.fingerprintPosition.indexOf("Index") != -1) {
          output[1] = finger;
        } else if (currFinger.fingerprintPosition.indexOf("Middle") != -1) {
          output[2] = finger;
        } else if (currFinger.fingerprintPosition.indexOf("Ring") != -1) {
          output[3] = finger;
        } else if (currFinger.fingerprintPosition.indexOf("Little") != -1) {
          output[4] = finger;
        } else {
          output.push(finger);
        }
      } else if (selected.type === "Palm") {
        if (
          currFinger.fingerprintPosition.indexOf("Lower") !== -1 &&
          currFinger.fingerprintPosition.indexOf("Palm") !== -1
        ) {
          output[0] = finger;
        } else if (
          currFinger.fingerprintPosition.indexOf("Writers") !== -1 &&
          currFinger.fingerprintPosition.indexOf("Palm") !== -1
        ) {
          output[1] = finger;
        } else {
          output.push(finger);
        }
      } else {
        output.push(finger);
      }
    });
  }

  hand = output;

  const fingerprintPositions = ["Thumb", "Index", "Middle", "Ring", "Little"];
  const palmPrintPositions = ["Lower", "Writers"];

  const outputX = [...output];
  if (selected.type !== "Palm") {
    for (let i = 0; i <= 4; i++) {
      if (outputX[i] === "undefined" || !outputX[i]) {
        outputX[i] = {
          axis: "",
          dateCreated: "",
          encoding: "",
          fingerprintPosition: `${selected.hand} ${fingerprintPositions[i]}`,
          fingerprintType: "",
          image: "",
          // impression: finger.impression,
          matchPersonTag: "",
          matchedFingerprintID: "",
          score: "",
          thumbNail: "",
          // annotation: finger.annotation,
          annotationNote: "",
          na: true
        };
      }
    }
  }
  // in case of palm prints, do this.
  else {
    for (let i = 0; i <= 1; i++) {
      // console.log("outputX[i]", outputX[i]);
      if (outputX[i] === "undefined" || !outputX[i]) {
        outputX[i] = {
          axis: "",
          dateCreated: "",
          encoding: "",
          fingerprintPosition: `${selected.hand} ${palmPrintPositions[i]}`,
          fingerprintType: "",
          image: "",
          // impression: finger.impression,
          matchPersonTag: "",
          matchedFingerprintID: "",
          score: "",
          thumbNail: "",
          // annotation: finger.annotation,
          annotationNote: "",
          na: true
        };
      }
    }
  }

  return outputX;
};

/**
 *
 * @param {any[]} matches
 * @param {any[]} related
 */
export const addMissingFinger = (matches, related) => {
  if (matches.length !== related.length) {
    related.map((finger, index) => {
      if (
        finger.annotation !== "NotAnnotated" &&
        finger.annotation !== "" &&
        !finger.thumbNail
      ) {
        matches.push({
          axis: "",
          dateCreated: "",
          encoding: finger.encoding,
          fingerprintPosition: finger.position,
          fingerprintType: finger.type,
          image: "",
          // impression: finger.impression,
          matchPersonTag: "",
          matchedFingerprintID: "",
          score: "",
          thumbNail: "",
          // annotation: finger.annotation,
          annotationNote: finger.annotationNote
        });
      }
    });
  }
  return matches;
};

export const addMissingFingerActual = (data, selected) => {
  const output = [];

  const first = data.find(item => item.position.indexOf("Thumb") !== -1);
  const second = data.find(item => item.position.indexOf("Index") !== -1);
  const third = data.find(item => item.position.indexOf("Middle") !== -1);
  const fourth = data.find(item => item.position.indexOf("Ring") !== -1);
  const fifth = data.find(item => item.position.indexOf("Little") !== -1);

  const fingerProps = name => {
    return {
      type: "Finger",
      position: `${selected.hand} ${name}`,
      impression: selected.type
    };
  };

  if (!first) {
    output.push(fingerProps("Thumb"));
  } else {
    output.push(first);
  }
  if (!second) {
    output.push(fingerProps("Index"));
  } else {
    output.push(second);
  }
  if (!third) {
    output.push(fingerProps("Middle"));
  } else {
    output.push(third);
  }
  if (!fourth) {
    output.push(fingerProps("Ring"));
  } else {
    output.push(fourth);
  }
  if (!fifth) {
    output.push(fingerProps("Little"));
  } else {
    output.push(fifth);
  }

  return output;
};
