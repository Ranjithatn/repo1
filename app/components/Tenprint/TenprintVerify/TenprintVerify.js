import React, { Fragment } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import Image from "../../Image/Image";
import WorkflowWrapper from "../../WorkflowWrapper/VerifierWorkflowWrapper";
import Button from "../../Button/Button";
import Title from "../../Title/Title";
import TenprintVerifyLeft from "./TenprintVerifyLeft/TenprintVerifyLeft";
import TenprintVerifyMain from "./TenprintVerifyMain/TenprintVerifyMain";
import TenprintVerifyTable from "./TenprintVerifyTable/TenprintVerifyTable";

import "./TenprintVerify.scss";

import {
  tenprintVerifyLeftFingersSelector,
  tenprintVerifyRightFingersSelector,
  tenprintVerifyPersonInfosSelector,
  tenprintVerifySelectedHandInfoSelector,
  tenprintVerifyLeftMatchInfoSelector,
  tenprintVerifyRightMatchInfoSelector,
  tenprintVerifyUserMatchInfoSelector,
  showAdjudicatorSelector,
  adjudicatorPassImageDataSelector,
  SelectedRowSelector,
  SelectedMatchSelector,
  adjudicatorImageDataSelector,
  MatchedUserRowIDSelector,
  tenprintVerifyRaw,
  adjudicatorResultSelector,
  databaseSelector,
  panesSelector
} from "../../../selectors/tenprint";
import {
  requestSetTenprintVerifyHand,
  requestSetTenprintVerifyMatchData,
  requestShowAdjudicator,
  requestMatchedPersonChanged,
  requestModalData,
  requestHideAdjudicator,
  requestTogglePaneVisibility
} from "../../../actions/tenprint";
import { requestShowModal } from "../../../actions/modal";
import { setTimeout } from "timers";
import {
  SelectedActionStatusSelector,
  SelectedActionTypeSelector,
  jobsTypeSelector
} from "../../../selectors/jobs";


import {
  // requestSelectedBiometrics,
  requestBiometricMugshot,
  requestScannedImageModal,
} from "../../../actions/scannedBiometrics";

// TENPRINT_VERIFY_CONFIRM
// let regex = /($[a-z])|[A-Z][^A-Z]+/g;
function TenprintVerify({
  formatMessage,
  history,
  LeftFingers,
  RightFingers,
  PersonInfos,
  requestSetTenprintVerifyHand,
  SelectedHandInfo,
  LeftMatches,
  RightMatches,
  MatchInfos,
  requestSetTenprintVerifyMatchData,
  requestShowAdjudicator,
  showAdjudicatorData,
  requestShowModal,
  SelectedRow,
  selectedMatch,
  ImageData,
  adjudicatorPassImageData,
  requestMatchedPersonChanged,
  requestModalData,
  ActionStatus,
  ActionType,
  requestHideAdjudicator,
  matchedRowID,
  tenprint,
  adjudicator,
  database,
  panes,
  requestTogglePaneVisibility,
  requestBiometricMugshot,
  requestScannedImageModal,
  jobType,
}) {
  // console.log("TenprintVerify SelectedHandInfo", SelectedHandInfo);
  // console.log("TenprintVerify:::jobType", jobType);

  // console.log("history is mystery", history);

  const routeChangeListener = history.listen(location => {
    // console.log("location", location);
    if (location.pathname != "/authenticated/tenprint/tenprintverify") {
      // console.log("route changed location", location);
      requestTogglePaneVisibility({ pane: "all", visible: true });
      routeChangeListener();
    }
  });

  return (
    <WorkflowWrapper
      className="tenprintverify-main"
      panes={panes}
      top={
        <Fragment>
          <div className="left">
            <Title
              is="6"
              text={`${formatMessage({ id: "search" })} : ${ database ? formatMessage({ id: database}) : "" }
                   | ${formatMessage({ id: "jobType" })} : ${ jobType ? formatMessage({ id: jobType}) : "" }`}
              // text={
              //   database
              //     ? formatMessage({ id: "search" }) + " : " + formatMessage({ id: database })
              //     : formatMessage({ id: "search" }) + " : "
              // }
            />
          </div>
          <div className="center">
            {ActionStatus === "Completed" ? (
              <Title is="4" text={formatMessage({ id: "statusCompleted" })} />
            ) : (
              ""
            )}
          </div>
          <div className="right">
            <Button
              leftIcon={panes.sidebar ? "expand" : "arrows-alt"}
              className="is-primary"
              onClick={() => {
                if (panes.sidebar) {
                  requestTogglePaneVisibility({ pane: "all", visible: false });
                } else {
                  requestTogglePaneVisibility({ pane: "all", visible: true });
                }
              }}
            />&nbsp;&nbsp;&nbsp;&nbsp;
            <Button
              text={formatMessage({ id: "back" })}
              className="is-primary"
              leftIcon="arrow-left"
              onClick={() => {
                if (!showAdjudicatorData) {
                  history.push("/authenticated/jobqueue");
                } else {
                  requestSetTenprintVerifyHand({ position: tenprint.selected });
                  requestTogglePaneVisibility({ pane: "all", visible: true });
                }
              }}
              // onClick={() => history.push("/authenticated/jobqueue")}
            />&nbsp;&nbsp;&nbsp;&nbsp;
            <Button
              text={formatMessage({ id: "next" })}
              className="is-primary"
              disabled={ActionStatus === "Completed"}
              rightIcon="arrow-right"
              onClick={() => requestModalData()}
            />
          </div>
        </Fragment>
      }
      left={
        <TenprintVerifyLeft
          formatMessage={formatMessage}
          LeftFingers={LeftFingers}
          RightFingers={RightFingers}
          requestSetTenprintVerifyHand={requestSetTenprintVerifyHand}
          SelectedHandInfo={SelectedHandInfo}
          MatchInfos={MatchInfos}
          requestShowAdjudicator={requestShowAdjudicator}
          adjudicatorPassImageData={adjudicatorPassImageData}
          LeftMatches={LeftMatches}
          RightMatches={RightMatches}
          SelectedRow={SelectedRow}
          requestHideAdjudicator={requestHideAdjudicator}
          tenprint={tenprint}
          panes={panes}
          requestTogglePaneVisibility={requestTogglePaneVisibility}
        />
      }
      main={
        <Fragment>
          <TenprintVerifyMain
            formatMessage={formatMessage}
            LeftFingers={LeftFingers}
            RightFingers={RightFingers}
            SelectedHandInfo={SelectedHandInfo}
            MatchInfos={MatchInfos}
            showAdjudicatorData={showAdjudicatorData}
            LeftMatches={LeftMatches}
            RightMatches={RightMatches}
            SelectedRow={SelectedRow}
            selectedMatch={selectedMatch}
            ImageData={ImageData}
            requestShowAdjudicator={requestShowAdjudicator}
            PersonInfos={PersonInfos}
            tenprint={tenprint}
            adjudicator={adjudicator}
            matchedRowID={matchedRowID}
            panes={panes}
            requestTogglePaneVisibility={requestTogglePaneVisibility}
            // adjudicatorPassImageData={adjudicatorPassImageData}
            requestBiometricMugshot={requestBiometricMugshot}
            requestScannedImageModal={requestScannedImageModal}
          />
          <TenprintVerifyTable
            formatMessage={formatMessage}
            PersonInfos={PersonInfos}
            LeftMatches={LeftMatches}
            RightMatches={RightMatches}
            SelectedHandInfo={SelectedHandInfo}
            requestSetTenprintVerifyMatchData={
              requestSetTenprintVerifyMatchData
            }
            MatchInfos={MatchInfos}
            SelectedRow={SelectedRow}
            requestShowModal={requestShowModal}
            requestMatchedPersonChanged={requestMatchedPersonChanged}
            ActionStatus={ActionStatus}
            matchedRowID={matchedRowID}
            tenprint={tenprint}
            requestSetTenprintVerifyHand={requestSetTenprintVerifyHand}
            panes={panes}
            requestTogglePaneVisibility={requestTogglePaneVisibility}
            database={database}
          />
        </Fragment>
      }
      foot={<div>Footer</div>}
    />
  );
}

const mapState = state => ({
  LeftFingers: tenprintVerifyLeftFingersSelector(state),
  RightFingers: tenprintVerifyRightFingersSelector(state),
  PersonInfos: tenprintVerifyPersonInfosSelector(state),
  SelectedHandInfo: tenprintVerifySelectedHandInfoSelector(state),
  LeftMatches: tenprintVerifyLeftMatchInfoSelector(state),
  RightMatches: tenprintVerifyRightMatchInfoSelector(state),
  MatchInfos: tenprintVerifyUserMatchInfoSelector(state),
  showAdjudicatorData: showAdjudicatorSelector(state),
  adjudicatorPassImageData: adjudicatorPassImageDataSelector(state),
  selectedMatch: SelectedMatchSelector(state),
  SelectedRow: SelectedRowSelector(state),
  ImageData: adjudicatorImageDataSelector(state),
  ActionStatus: SelectedActionStatusSelector(state),
  ActionType: SelectedActionTypeSelector(state),
  matchedRowID: MatchedUserRowIDSelector(state),
  tenprint: tenprintVerifyRaw(state),
  adjudicator: adjudicatorResultSelector(state),
  database: databaseSelector(state),
  panes: panesSelector(state),
  jobType: jobsTypeSelector(state),
});

export default connect(mapState, {
  requestSetTenprintVerifyHand,
  requestSetTenprintVerifyMatchData,
  requestShowAdjudicator,
  requestShowModal,
  requestMatchedPersonChanged,
  requestModalData,
  requestHideAdjudicator,
  requestTogglePaneVisibility,
  requestBiometricMugshot,
  requestScannedImageModal,
})(withRouter(TenprintVerify));
