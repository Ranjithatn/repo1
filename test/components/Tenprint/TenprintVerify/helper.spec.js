import React from "react";
import toJson from "enzyme-to-json";
import { findAllFingers } from "../../../../app/components/Tenprint/TenprintVerify/Helper"
import { shallow, configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

let component, tree, instance;


let props = {
    biometrics:{}, hand:{}, type:{}
};

// beforeEach(() => {
//   component = mount(<NewActionModal {...props} />);
//   criminalDatabaseToggle = component.find("input#searchCriminalDB");
//   civilDatabaseToggle = component.find("input#searchCivilDB");
//   latentDatabaseToggle = component.find("input#searchLatentDB");
//   // registerCriminalToggle = component.find("input#registerCriminal");
//   // editCriminalToggle = component.find("input#editCriminalData");
//   // updateCriminalToggle = component.find("input#updateCriminalFp");
//   // deleteCriminalToggle = component.find("input#deleteCriminalFp");
//   // visualVerifyToggle = component.find("input#visualVerify");
//   tree = toJson(component);
//   instance = component.instance();
// });

describe("<findAllFingers />", () => {
  it("should render without crashing", () => {
    expect(tree).toMatchSnapshot();
  });
 
});
