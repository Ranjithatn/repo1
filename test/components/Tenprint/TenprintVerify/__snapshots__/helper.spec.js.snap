// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<findAllFingers /> should render without crashing 1`] = `undefined`;
